scripts/configure-venue-access.ts

```diff
@@ -0,0 +1,141 @@
+import fs from "fs";
+import admin from "firebase-admin";
+import serviceAccount from "./sparkle-7-firebase-adminsdk-f5a3m-482bebb057.json";
```

Comment:
This shouldn't be merged like this. We have patterns for correctly loading the required keys into our helper scripts in `scripts/lib/helpers`. See also the patterns such as `initFirebaseAdminApp` used in the most recently updated scripts, such as those in https://github.com/sparkletown/sparkle/pull/1065

Reply:
Is this supposed to come from the database, .env file or something else? I don't know how this is supposed to work, I need some help here.

Reply:
As far as getting that file locally on your machine, you would follow the steps in https://github.com/sparkletown/sparkle/blob/staging/scripts/README.md as normal.

My comments above are specifically around how we import it into this script/etc.

If you look at `scripts/get-badges` in https://github.com/sparkletown/sparkle/pull/1065/files#diff-d9419eb52fa2dc609fd49dccb22b7a2f2a0df95afcbbc5aaab56a587110974e3 for example, you can see that there is no `serviceAccount` import at all.

```typescript
initFirebaseAdminApp(projectId, {
  credentialPath: credentialPath
    ? resolve(__dirname, credentialPath)
    : undefined,
});
```

It allows the path to the key to be passed via the CLI params (`credentialPath`), or when that is `undefined`, `initFirebaseAdminApp` will try to load the default serviceAccount (see the definition in `scripts/lib/helpers` for specifics):

https://github.com/sparkletown/sparkle/blob/b01f80af73a4e1dc28467f85201a3ed6c4fa3532/scripts/lib/helpers.ts#L22-L28

---

@denisdimitrov does that answer your questions/unblock you here?

Reply:
captured the above on the documentation issue at https://github.com/sparkletown/sparkle/pull/1062#issuecomment-763287172